### PR TITLE
Build friendly url

### PR DESCRIPTION
**Pull request by Danwhy**
ref #296

If only one issue tag is selected, makes a friendly url which is the landing page + '/' + the issue tag name. If multiple urls selected, builds a url with a query string.

### CODE REVIEW

File: cms/static/js/personalise.js [hunk 1]
```diff
@@ -42,6 +42,11 @@ function getQuery() {
       query[a].push(myArray[1].split(splitreg).join(' '));
     }
   });
+
+  if(window.location.href.split('/').length == 6) {
+    query['issue'] = [window.location.href.split('/')[4].split("-").join(" ")];
```
Comment (samhstn) [inline cms/static/js/personalise.js:47]: `==` -> `===`

`.split("-").join(" ")` -> `.replace("-", " ")`

File: cms/static/js/personalise.js [hunk 2]
```diff
@@ -152,28 +157,71 @@ function getOrder() {
...
+    tags[el] = getTagsOfType(el);
+  });
+
+  if (
+    tags.issue.length === 0 && !tagSelected(tag) &&
+    tags.reason.length === 0 && tags.content.length === 0 && tag.tag_type === 'issue'
+  ) {
+    /* No tags are selected & this is an issue tag - make friendly url */
+    querystring = tag.name.split(' ').join('-');
+  } else if (tags.issue.length == 1 && tagSelected(tag) && tags.reason.length === 0 && tags.content.length === 0) {
```
Comment (samhstn) [inline cms/static/js/personalise.js:175]: `==` 🚩

File: resources/models/home.py [hunk 3]
```diff
@@ -358,7 +376,11 @@ def custom_serve(self, request, *args, **kwargs):
 
     project_info_block = ProjectInfoBlock.objects.all()
 
-    context = self.get_context(request)
+    path_components = kwargs.get('path_components', [])
+
+    list(map(lambda x: " ".join(x.split("-")), path_components))
```
Comment (samhstn) [inline resources/models/home.py:381]: There's also a python [String replace()](https://www.tutorialspoint.com/python/string_replace.htm) which might be nicer

**Review by samhstn — APPROVED** [review]
Sorry for the petty points, looks good

File: cms/static/js/personalise.js [hunk 4]
```diff
@@ -152,28 +157,71 @@ function getOrder() {
...
+  ['issue', 'reason', 'content'].forEach(function(el) {
+    tags[el] = getTagsOfType(el);
+  });
+
+  if (
+    tags.issue.length === 0 && !tagSelected(tag) &&
+    tags.reason.length === 0 && tags.content.length === 0 && tag.tag_type === 'issue'
+  ) {
+    /* No tags are selected & this is an issue tag - make friendly url */
+    querystring = tag.name.split(' ').join('-');
```
Comment (SimonLab) [inline cms/static/js/personalise.js:174]: replace here also?

File: cms/static/js/personalise.js [hunk 5]
```diff
@@ -152,28 +157,71 @@ function getOrder() {
...
+    matches.push(res[1]);
   }
 
-  history.replaceState(null, null, newUrl);
+  return matches;
 }
 
 function tagSelected(tag) {
   var jointTag = tag.name.split(' ').join('%20');
+  var hyphenTag = tag.name.split(' ').join('-');
```
Comment (SimonLab) [inline cms/static/js/personalise.js:222]: ```replace```

**Review by SimonLab — COMMENTED** [review]
looks good, some small enhancement but again not a blocker. @Danwhy let me know if you want to do it on this PR or next a next one

**Comment by Danwhy** [comment]
:+1: I'll make those changes in #356

**Review by SimonLab — APPROVED** [review]

**Pull request merged**